sbol_utilities/sbol3_genbank_conversion.py

```diff
@@ -586,3 +510,157 @@ def _reset_extra_properties_in_genbank(self, obj: sbol3.Component, seq_rec: SeqR
+                    # Create a cut or range as featurelocation depending on whether location is specified as
+                    # Cut (eg: "n^n+1", parsed as [n:n] by biopython) or Range (eg: "n..m", parsed as [n:m] by biopython)
+                    if gb_loc.start == gb_loc.end:
+                        locs = sbol3.Cut(
+                            sequence=seq,
+                            at=int(gb_loc.start),
+                            orientation=feat_loc_orientation,
+                        )
+                    else:
+                        locs = sbol3.Range(
+                            sequence=seq,
+                            start=int(gb_loc.start),
+                            end=int(gb_loc.end),
+                            orientation=feat_loc_orientation,
+                        )
+                    feat_locations.append(locs)
+                # Obtain sequence feature role from gb2so mappings
+                feat_role = sbol3.SO_NS[:-3]
+                if self.gb2so_map.get(gb_feat.type):
+                    feat_role += self.gb2so_map[gb_feat.type]
+                else:
+                    logging.warning(f"Feature type: `{gb_feat.type}` for feature: `{gb_feat.qualifiers['label'][0]}` \n \
+                    of record: `{record.name}` has no corresponding ontology term for SO, using the default SO term, {self.DEFAULT_SO_TERM}")
+                    feat_role += self.DEFAULT_SO_TERM
+                # assign feature orientation based on the strand value in genbank feature
+                feat_orientation = sbol3.SO_FORWARD
+                if gb_feat.strand == -1:
+                    feat_orientation = sbol3.SO_REVERSE
+                feat = self.Feature_GenBank_Extension(
+                    locations=feat_locations,
+                    roles=[feat_role],
+                    # name=gb_feat.qualifiers["label"][0],
+                    name=feat_name,
+                    orientation=feat_orientation
+                )
+                # store qualifiers key value pairs
+                for ind, qualifier in enumerate(gb_feat.qualifiers):
+                    feat.qualifier_key.append(f"{ind}:" + qualifier)
+                    feat.qualifier_value.append(f"{ind}:" + gb_feat.qualifiers[qualifier][0])
+                comp.features.append(feat)
+
+
+    def _handle_features_sbol_to_gb(self, seq_rec: SeqRecord, obj: sbol3.Component, seq_rec_features: List) -> None:
+        """Helper function for resetting sequence features and their qualifiers to GenBank,
+        by using a modified, extended SBOL3 Sequence Feature class - Feature_GenBank_Extension.
+        :param seq_rec: GenBank SeqRecord instance for the record which contains sequnce features
+        :param obj: Instance of the SBOL3 Component 
+        :param seq_rec_features: List of features for the related sequence features
+        """
+        seq_rec_features = []
+        if obj.features:
```

Comment:
Note: I am not asking for a change, I am making an observation.

The same applies here as in `_handle_features_gb_to_sbol`. This method could short-circuit the case where there are no `obj.features`. It is hard for me to tell if the last two lines of the method align with this `if` statement, or if they are indented inside of the `if` statement.

Reply:
@tcmitchell  same here, have made this change as it does improve the readability, thanks for pointing it out